### PR TITLE
Lint: Migration to Stylistic

### PR DESCRIPTION
**Pull request by TruongQuangSB**
- Use rules from https://eslint.style/guide/why
- Eslint: Migration to flat configuration files

### CODE REVIEW

File: web/siteplan/src/util/MapSources/MapboxTop.ts [hunk 1]
```diff
@@ -32,7 +31,7 @@ interface MapboxStyle
  * @author Stuecker
  */
 export default class MapboxTop extends MapSource {
-  source: VectorTileSource | null = null
+  source: VectorTileSource|null = null
```
Comment (mariusheine) [inline web/siteplan/src/util/MapSources/MapboxTop.ts:34]: @TruongQuangSB Weird formatting. I would keep the spaces

File: web/siteplan/src/util/MapSources/MapboxTop.ts [hunk 2]
```diff
@@ -76,21 +75,21 @@ export default class MapboxTop extends MapSource {
   }
 
   onStyleLoad (style: MapboxStyle): void {
-    const sourceId = Object.keys(style.sources)[0]
+    const sourceId = Object.keys(style.sources)[ 0 ]
```
Comment (mariusheine) [inline web/siteplan/src/util/MapSources/MapboxTop.ts:78]: @TruongQuangSB Here I would stay with no spaces around the `0`

File: web/textviewer/package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "@typescript-eslint/eslint-plugin": "^8.11.0",
     "@typescript-eslint/parser": "^8.11.0",
     "copy-webpack-plugin": "^12.0.2",
-    "eslint": "^8.57.1",
+    "@stylistic/eslint-plugin": "2.8.0",
```
Comment (mariusheine) [inline web/textviewer/package.json:21]: @TruongQuangSB Same here as in package.json of siteplan

File: web/siteplan/.eslintrc.cjs [hunk 4]
```diff
@@ -15,16 +15,17 @@ module.exports = {
     ecmaVersion: '2020'
   },
   rules: {
-    'no-console': process.env.NODE_ENV === 'production' ? ['warn', { allow: ['warn', 'error'] }] : 'off',
+    'no-console': process.env.NODE_ENV === 'production' ? ['warn', {allow: ['warn', 'error']}] : 'off',
```
Comment (mariusheine) [inline web/siteplan/.eslintrc.cjs:18]: @TruongQuangSB Here I would stay with spaces...

**Review by mariusheine — CHANGES_REQUESTED** [review]

File: web/siteplan/package.json [hunk 5]
```diff
@@ -28,9 +28,9 @@
     "@types/geojson": "^7946.0.14",
     "@types/proj4": "^2.5.2",
     "@typescript-eslint/eslint-plugin": "^8.11.0",
+    "@stylistic/eslint-plugin": "2.8.0",
     "@typescript-eslint/parser": "^8.11.0",
     "@vitejs/plugin-vue": "^5.1.4",
-    "eslint": "^9.12.0",
```
[the remark this answers is not in the document]
Comment (TruongQuangSB) [reply web/siteplan/package.json:33]: @mariusheine here used the `stylistic` as plugins to lint parse and this package already contained `eslint`

[the remark this answers is not in the document]
Comment (mariusheine) [reply web/siteplan/package.json:33]: @TruongQuangSB But it only has it as peer dependencies... https://www.npmjs.com/package/@stylistic/eslint-plugin?activeTab=dependencies

When you look at their github repo you see that it has it only as peer dependency: https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/package.json

that's why I am pretty confused...

[the remark this answers is not in the document]
Comment (TruongQuangSB) [reply web/siteplan/package.json:33]: ok, u are right

File: web/textviewer/src/app.ts [hunk 6]
```diff
@@ -22,7 +22,7 @@ export class App {
   editor: monaco.editor.IStandaloneCodeEditor
   problems: ProblemMessage[]
   xml!: Document
-  pendingSetLine: number | null = null
+  pendingSetLine: number|null = null
```
Comment (mariusheine) [inline web/textviewer/src/app.ts:25]: @TruongQuangSB At other locations we have spaces between the pipe

**Review by mariusheine — APPROVED** [review]
@TruongQuangSB One last thing :smile:

File: web/textviewer/src/app.ts (hunk 6 above)
Comment (TruongQuangSB) [reply web/textviewer/src/app.ts]: @mariusheine it given't rule no space before and after the pipe by type annotation. We must check manuael 😄

**Review by mariusheine — APPROVED** [review]

**Pull request merged**